R/add_stylecss.R

```diff
@@ -0,0 +1,40 @@
+#' add_stylecss
+#' @description Add PHS template style.css file to chosen directory.
+#'
+#' @param path String: path to add style.css file. If left blank, will prompt the user within R Studio
+#'
+#' @return Adds style.css file to directory.
+#' @export
+#' @examples
+#' \dontrun{
+#' add_stylecss()
+#' }
+add_stylecss <- function(path = rstudioapi::selectDirectory(caption = "Select folder to add style.css")) {
+  if (is.null(path)) {
+    return(message("style.css file not added."))
+  }
+
+  # style.css content to add
+  stylecss <- readLines(system.file(package = "phstemplates", "text", "stylecss.txt"))
+
+  # collect into single text string
+  stylecss <- paste(stylecss, collapse = "\n")
+
+  # Search for existing stylecss in path
+  file_exists <- file.exists(paste0(path, "/style.css"))
+
+  if (file_exists) {
+    write_out <- rstudioapi::showQuestion(
+      title = "Overwrite?",
+      message = "You already have a style.css file. Select OK if you want to overwrite this file?",
+    )
+  } else {
+    write_out <- TRUE
+  }
+
+  if (write_out) {
+    writeLines(stylecss, con = paste0(path, "/style.css"))
+  } else {
+    return(message("style.css file not added."))
+  }
```

Comment:
Hopefully, this re-write looks good to you - using `file.path` instead of `paste0` and then refactoring to cut down code.

```suggestion
  if (file.exists(file.path(path, "style.css"))) {
    overwrite <- rstudioapi::showQuestion(
      title = "Overwrite?",
      message = "A style.css file already exists. Select OK to overwrite this file."
    )
  
    if (!overwrite) {
      message("style.css has not been overwritten.")
      return(NULL)
    }
  }
  
  writeLines(stylecss, con = file.path(path, "style.css"))
  return(NULL)
```

Reply:
I think the object `overwrite` won't exist if the file doesn't already exist? So we need an `else` to make sure it's created?

Reply:
Line 30 is nested within the other if, so will only execute if the file doesn't exist.

Reply:
ah thanks, my eyes couldn't see the nesting for some reason!